custom/icds/data_management/doc_processors.py

```diff
@@ -106,6 +111,39 @@ def should_process(self, doc):
         return False
 
 
+class SanitizeFemaleDeathTypeDocProcessor(DataManagementDocProcessor):
+    """Sanitize: Reset female_death_type attribute to "" (empty):
+       If the person record is MALE.
+       (OR)
+       If `died` attribute is not `yes`.
+
+    Args:
+        domain: Domain reference
+    """
+    def __init__(self, domain):
+        super().__init__(domain)
+        self.test_location_ids = find_test_awc_location_ids(self.domain)
+
+    def process_bulk_docs(self, docs, progress_logger):
+        if docs:
+            updates = {doc['_id']: {FEMALE_DEATH_TYPE_PROPERTY: ''} for doc in docs}
+            for doc in docs:
+                progress_logger.document_processed(doc, updates[doc['_id']])
+            submit_case_blocks(self._create_case_blocks(updates),
+                               self.domain, user_id=SYSTEM_USER_ID)
+        return True
+
+    def should_process(self, doc):
+        owner_id = doc.get('owner_id')
+        if owner_id and owner_id in self.test_location_ids:
+            return False
+        if doc.get(SEX_PROPERTY) == SEX_PROPERTY_MALE_VALUE:
+            return bool(doc.get(FEMALE_DEATH_TYPE_PROPERTY))
+        if doc.get(HAS_DIED_PROPERTY) != HAS_DIED_PROPERTY_YES_VALUE:
+            return bool(doc.get(FEMALE_DEATH_TYPE_PROPERTY))
```

Comment:
this might be easier to read as
```
if doc.get(FEMALE_DEATH_TYPE_PROPERTY):
    if doc.get(SEX_PROPERTY) == SEX_PROPERTY_MALE_VALUE or doc.get(HAS_DIED_PROPERTY) != HAS_DIED_PROPERTY_YES_VALUE:
        return True
```